pkg/reporting/pullRequestReport.go

```diff
@@ -0,0 +1,292 @@
+						VulnerabilityHref:  policyVulnerability.Meta.Href,
+						VulnerabilityScore: getScore(policyVulnerability.ErrorMessage, "score") + " " + getScore(policyVulnerability.ErrorMessage, "severity"),
+						ComponentName:      componentName,
+					})
+			}
+		}
+
+	}
+	// sorting allPolicyViolations
+	sort.Slice(allPolicyViolations, func(i, j int) bool {
+		return allPolicyViolations[i] < allPolicyViolations[j]
+	})
+	// sorting componentNames
+	sort.Slice(componentNames, func(i, j int) bool {
+		return componentNames[i] < componentNames[j]
+	})
+
+	if allPolicyViolations != nil && componentNames != nil {
+		scanReport.Success = false
+		// MainTableHeaders
+		scanReport.MainTableHeaders = append(scanReport.MainTableHeaders, "Component name")
+		for _, policyViolationName := range allPolicyViolations {
+			scanReport.MainTableHeaders = append(scanReport.MainTableHeaders, policyViolationName)
+		}
+		// MainTableValues
+		for i := range componentNames {
+			scanReport.MainTableValues = append(scanReport.MainTableValues, []string{})
+			scanReport.MainTableValues[i] = append(scanReport.MainTableValues[i], componentNames[i])
+			for j := 1; j < len(scanReport.MainTableHeaders); j++ {
+				policyV := scanReport.MainTableHeaders[j]
+				comp := componentNames[i]
+				count := strconv.Itoa(countPolicyViolationComponent[policyV][comp])
+				scanReport.MainTableValues[i] = append(scanReport.MainTableValues[i], count)
+			}
+		}
+		// sorting VulnsTableValues
+		for _, item := range scanReport.VulnsTableValues {
+			sort.Slice(item, func(i, j int) bool {
+				if strings.Contains(item[i].VulnerabilityScore, "10.0") {
+					return true
+				} else if strings.Contains(item[j].VulnerabilityScore, "10.0") {
+					return false
+				}
+				if item[i].VulnerabilityScore > item[j].VulnerabilityScore {
+					return true
+				} else if item[i].VulnerabilityScore < item[j].VulnerabilityScore {
+					return false
+				}
+				return item[i].ComponentName < item[j].ComponentName
+			})
+		}
+		// sorting NotVulnsTableHeaders
+		for _, item := range scanReport.NotVulnsTableValues {
+			sort.Slice(item, func(i, j int) bool {
+				return item[i].ComponentName <= item[j].ComponentName
+			})
+		}
+
+	}
+
+	tmpl, err := template.New("report").Parse(rapidReportMdTemplate)
+	if err != nil {
+		return "", errors.New("failed to create Markdown report template err:" + err.Error())
+	}
+	buf := new(bytes.Buffer)
+	err = tmpl.Execute(buf, scanReport)
+	if err != nil {
+		return "", errors.New("failed to create Markdown report template err:" + err.Error())
+	}
+	// removing report files
+	err = os.RemoveAll(removeDir)
+	if err != nil {
+		log.Println("couldn't remove report file")
+	}
+	return buf.String(), nil
+}
+
+type Files []os.DirEntry
+
+func (a Files) Len() int           { return len(a) }
+func (a Files) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a Files) Less(i, j int) bool { return a[i].Name() < a[j].Name() }
```

Comment:
I think is is better to just find last created file because it has less complexity than sorting files and we only need the last created directory. The same way it was done in repository which creates pull requests comments using GitHub actions.

```suggestion
func findLastCreatedDir(directories []os.DirEntry) os.DirEntry {
	lastCreatedDir := directories[0]
	for _, dir := range directories {
		if dir.Name() > lastCreatedDir.Name() {
			lastCreatedDir = dir
		}
	}

	return lastCreatedDir
}
```